### ISSUE TITLE
[Feature Request] Playback Volume Normalization

### ISSUE THREAD

**Issue by SunBreaker1** [zyrouge/symphony#57]
Samsung's music app is the only app that I know of that has the feature to normalize the volume of all songs being played to the same volume. 

Samsung's implementation: "Smart Volume - automatically adjust each track to the same volume".

This would be a pretty cool Quality of Life feature so that you don't have a loud song play right after a quiet song and blow your ear drums or speakers.

Hopefully this can be implemented without ruining the dynamic range or mastering of the song. From researching, a system based on LUFS would work best. 

An extra optional feature would be to allow the end user to set a custom LUFS volume output, some like it louder, some like it quieter, this would allow all users to customize to their preferences.

I'm not a programmer, so I don't know how easy or hard this would be to implement, I just think it would be a very useful feature.

**Comment by SunBreaker1** [comment]
I asked ChatGPT to "Write code of an audio player feature that adjusts the volume of songs to the same level using LUFS in Kotlin language" and to "Adjust the code to allow the user to change the LUFS value" and it seem that this is possible. 

But because I'm not a programmer, this goes miles above my head, so I don't know how feasible it is to implement. Its just fun to play with ChatGPT to see what code it can create.

**Comment by zyrouge** [comment]
I've never tried to do this, but I'll see the possibilities of implementing this!

**Comment by zyrouge** [comment]
Wouldn't supporting ReplayGain be a better idea?

**Comment by SunBreaker1** [comment]
> Wouldn't supporting ReplayGain be a better idea?

After suggesting this same feature request to JamesDSP, they did in fact suggest this. Alternatively, the suggested using other players, but I'd rather not since I've experimented with multiple and I found symphony to be my favorite. 

Sorry, getting back so late

**Comment by SunBreaker1** [comment]
> > Wouldn't supporting ReplayGain be a better idea?
> 
> After suggesting this same feature request to JamesDSP, they did in fact suggest this. Alternatively, the suggested using other players, but I'd rather not since I've experimented with multiple and I found symphony to be my favorite. 
> 
> Sorry, getting back so late

From my brief testing, the implementation of Replaygain in aimp is inconsistent. 

The main song I used; Bully - Calliope Mori and Replaygain 5db. The first few seconds of the song were unchanged and a little loud, then it slowly started to get quieter as the song progressed stabilizing at a lower volume. 

Tldr; starts loud then gets quieter 

I then tried with Replaygain off, and the volume stayed consistent throughout the song. Same thing happens with Boof Pack - Lord Aethelstan

**Comment by zyrouge** [comment]
Guess I'll have to wait for more feedback.

**Comment by SunBreaker1** [comment]
> Wouldn't supporting ReplayGain be a better idea?

I've done some research and experimenting, and yes this would definitely be the play. I tried out Foobar2000's replay gain after following Z Reviews vid on it, and tried it out on Foobar on desktop and pc and it works great. I would use Foobar on android but the interface is frankly outdated (imo) and has problems with albums.

If implementing ReplayGain the way Foobar has it into Symphony, that would easily make this app one of the top players (imo)
